examples/lambda-case.kl

```diff
@@ -0,0 +1,15 @@
+#lang "n-ary-app.kl"
+
+(import (shift "n-ary-app.kl" 1))
+(import (shift "quasiquote.kl" 1))
+
+(define-macros
+  ([lambda-case
+    (lambda (stx)
+      (syntax-case stx
+        [(list (_ cases))
+         (pure
+          (quasiquote/loc stx
+            (lambda (x) (case x ,cases))))]))]))
```

Comment:
It's unfortunate that we can't use our `define-syntax-rule` for such a simple macro! I guess we should add support for the `...` syntax in `quasiquote`.

Reply:
I don't think ... belongs as part of quasiquote, but a pattern language for syntax objects that's along the lines of Dybvig's syntax-case would sure be nice!